Microsoft.Toolkit.Uwp.SampleApp/SamplePages/OneDrive Service/OneDrivePage.xaml.cs

```diff
@@ -74,34 +76,53 @@ public OneDrivePage()
             catch (ServiceException serviceEx)
             {
                 await OneDriveSampleHelpers.DisplayOneDriveServiceExceptionAsync(serviceEx);
+                succeeded = false;
             }
             catch (Exception ex)
             {
                 await OneDriveSampleHelpers.DisplayMessageAsync(ex.Message);
                 TrackingManager.TrackException(ex);
+                succeeded = false;
             }
             finally
             {
                 Shell.Current.DisplayWaitRing = false;
             }
 
-            FilesBox.Visibility = Visibility.Visible;
-            UserBox.Visibility = Visibility.Visible;
-            ClientIdBox.Visibility = Visibility.Collapsed;
-            LogOutButton.Visibility = Visibility.Visible;
-            ConnectButton.Visibility = Visibility.Collapsed;
-            menuButton.Visibility = Visibility.Visible;
-            BackButton.Visibility = Visibility.Visible;
+            if (succeeded)
+            {
+                FilesBox.Visibility = Visibility.Visible;
+                UserBox.Visibility = Visibility.Visible;
+                ClientIdBox.Visibility = Visibility.Collapsed;
+                ClientIdHelper.Visibility = Visibility.Collapsed;
+                LogOutButton.Visibility = Visibility.Visible;
+                ConnectButton.Visibility = Visibility.Collapsed;
+                menuButton.Visibility = Visibility.Visible;
+                BackButton.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                FilesBox.Visibility = Visibility.Collapsed;
+                UserBox.Visibility = Visibility.Collapsed;
+                ClientIdBox.Visibility = Visibility.Visible;
+                ClientIdHelper.Visibility = Visibility.Visible;
+                LogOutButton.Visibility = Visibility.Collapsed;
+                ConnectButton.Visibility = Visibility.Visible;
+                menuButton.Visibility = Visibility.Collapsed;
+                BackButton.Visibility = Visibility.Collapsed;
+            }
         }
 
         private async void ConnectButton_Click(object sender, RoutedEventArgs e)
         {
-            if (string.IsNullOrEmpty(ClientId.Text))
+            try
             {
-                return;
+                await SigninAsync(_indexProvider, ClientId.Text);
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex.Message);
```

Comment:
Swallowing the exception and just debugging it seems a bit wrong, maybe adding a `MessageDialog` here also?